app/assets/javascripts/services/miq_service.js

```diff
@@ -75,6 +75,12 @@ ManageIQ.angular.app.service('miqService', ['$timeout', '$document', '$q', 'API'
     }, 200);
   };
 
+  this.detectWithRest = function($event, url) {
+    angular.element('#button_name').val('detect');
+    miqSparkleOn();
+    return miqRESTAjaxButton(url, $event.target, 'json');
```

Comment:
When wrapping a non-angular Promise returning function, it's a good idea to wrap the result in an angular promise - so that you don't need that `$scope.$apply` when calling it.

`return $q.when(miqRESTAjaxButton(url, $event.target, 'json'));` should work ;)